c/busybox-1.22.0/cal_false-unreach-call_true-no-overflow_true-valid-memsafety.c

```diff
@@ -1864,6 +1864,31 @@ int getopt_long(int argc, char * const argv[], const char *optstring,
   return result;
 }
 
+int vasprintf(char **ptr, const char *fmt, va_list ap)
+{
+  (void)*fmt;
+  (void)ap;
+
+  if(__VERIFIER_nondet_int())
+    return -1;
+
+  // limiting the resulting string to at most 42 characters is an
+  // underapproximation obviously
+  *ptr = malloc(42);
+  int i = 0;
+  while(i<42)
+  {
+    (*ptr)[i] = __VERIFIER_nondet_char();
+    if((*ptr)[i] == '0')
```

Comment:
Should this not be `'\0'` or maybe even better simply `0` instead of `'0'`?
I would guess that you are trying to terminate this string with `0` here.